src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/jobs/LicenceActivationServiceTest.kt

```diff
@@ -0,0 +1,304 @@
+  }
+
+  @Test
+  fun `licence activation job calls for HDC approved licences to be deactivated`() {
+    whenever(licenceRepository.getApprovedLicencesOnOrPassedReleaseDate()).thenReturn(listOf(aLicenceEntity))
+    whenever(prisonApiClient.hdcStatuses(listOf(54321))).thenReturn(listOf(anHdcStatus))
+
+    service.licenceActivationJob()
+
+    verify(licenceService, times(0)).activateLicences(anyList())
+    verify(licenceService, times(1)).inactivateLicences(listOf(aLicenceEntity))
+  }
+
+  @Test
+  fun `licence activation job calls for non-IS91 licences with HDC (not approved) to be activated on their ARD if the offender has been released`() {
+    whenever(licenceRepository.getApprovedLicencesOnOrPassedReleaseDate()).thenReturn(listOf(aLicenceEntity))
+    whenever(prisonerSearchApiClient.searchPrisonersByBookingIds(listOf(54321))).thenReturn(
+      listOf(prisonerSearchPrisoner)
+    )
+    whenever(prisonApiClient.getIS91AndExtraditionBookingIds(listOf(54321))).thenReturn(emptyList())
+    whenever(prisonApiClient.hdcStatuses(listOf(54321))).thenReturn(listOf(anHdcStatus.copy(approvalStatus = "INACTIVE")))
+
+    service.licenceActivationJob()
+
+    verify(licenceService, times(1)).activateLicences(listOf(aLicenceEntity))
+    verify(licenceService, times(0)).inactivateLicences(anyList())
+  }
+
+  @Test
+  fun `licence activation job does not call for non-IS91 licences to be activated if the offender has not been released`() {
+    whenever(licenceRepository.getApprovedLicencesOnOrPassedReleaseDate()).thenReturn(listOf(aLicenceEntity))
+    whenever(prisonerSearchApiClient.searchPrisonersByBookingIds(listOf(54321))).thenReturn(
+      listOf(
+        prisonerSearchPrisoner.copy(status = "ACTIVE IN")
+      )
+    )
+    whenever(prisonApiClient.getIS91AndExtraditionBookingIds(listOf(54321))).thenReturn(emptyList())
+    whenever(prisonApiClient.hdcStatuses(listOf(54321))).thenReturn(emptyList())
+
+    service.licenceActivationJob()
+
+    verify(licenceService, times(0)).activateLicences(anyList())
+    verify(licenceService, times(0)).inactivateLicences(anyList())
+  }
+
+  @Test
+  fun `licence activation job does not call for non-IS91 licences to be activated if on CRD and ARD is in the future`() {
+    whenever(licenceRepository.getApprovedLicencesOnOrPassedReleaseDate()).thenReturn(
+      listOf(
+        aLicenceEntity.copy(
+          actualReleaseDate = LocalDate.now().plusDays(1),
+          conditionalReleaseDate = LocalDate.now()
+        )
+      )
+    )
+    whenever(prisonerSearchApiClient.searchPrisonersByBookingIds(listOf(54321))).thenReturn(
+      listOf(
+        prisonerSearchPrisoner
+      )
+    )
+    whenever(prisonApiClient.getIS91AndExtraditionBookingIds(listOf(54321))).thenReturn(emptyList())
+    whenever(prisonApiClient.hdcStatuses(listOf(54321))).thenReturn(emptyList())
+
+    service.licenceActivationJob()
+
+    verify(licenceService, times(0)).activateLicences(anyList())
```

Comment:
rather than using times(0), I think you can pass `never()` as second param, or use `verifyNoMoreInteractions(licenceService)`.

but if you remove the unnecessary empty check then this could be verify passed an `emptyList()`